flytekit/remote/remote.py

```diff
@@ -367,28 +367,22 @@ def _resolve_identifier(self, t: int, name: str, version: str, ss: Serialization
     def _serialize_and_register(
         self,
         entity: FlyteLocalEntity,
-        settings: typing.Optional[SerializationSettings],
         version: str,
+        settings: typing.Optional[SerializationSettings] = None,
         options: typing.Optional[Options] = None,
     ) -> Identifier:
         """
         This method serializes and register the given Flyte entity
         :return: Identifier of the registered entity
         """
         m = OrderedDict()
-        # Create dummy serialization settings for now.
-        # TODO: Clean this up by using lazy usage of serialization settings in translator.py
-        serialization_settings = (
-            settings
-            if settings
-            else SerializationSettings(
-                ImageConfig.auto_default_image(),
-                project=self.default_project,
-                domain=self.default_domain,
-                version=version,
+
+        if settings is None:
+            settings = SerializationSettings(
```

Comment:
can we not have the dummy settings here?  it feels a little unclean - honestly the places where we already use dummy settings should be removed.  can we update the get_serializable so that it only looks for the SerializationSettings object when it's necessary?

Reply:
hmm, I will have to change the function signature and the default value. I cannot let it be `None` since there are further conditions that check for `None` inside `_serialize_and_register` beneath. The dummy settings are now a part of the default value and the API has been refactored for every affected file. Hope this helps!